OData/src/System.Web.OData/OData/Query/ODataQueryOptions.cs

```diff
@@ -508,7 +508,13 @@ public static IQueryable<T> LimitResults<T>(IQueryable<T> queryable, int limit,
             return truncatedCollection.AsQueryable();
         }
 
-        internal static Uri GetNextPageLink(HttpRequestMessage request, int pageSize)
+        /// <summary>
+        /// Creates a link for the next page of results; To be used as the value of @odata.nextLink.
+        /// </summary>
+        /// <param name="request">The request on which to base the next page link.</param>
+        /// <param name="pageSize">The number of results allowed per page.</param>
+        /// <returns>A next page link.</returns>
+        public static Uri GetNextPageLink(HttpRequestMessage request, int pageSize)
         {
             Contract.Assert(request != null);
```

Comment:
Thanks @brjohnstmsft. We need to check if the `request` is not null, and also the `request.RequestUri` is not null etc. for a public API; and add a few test cases.